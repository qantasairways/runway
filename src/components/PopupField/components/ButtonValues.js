import React from 'react';
import PropTypes from 'prop-types';

import { fontSize, colours, letterSpacing } from '../../../theme/airways';

function ButtonValue({ large, small }) {
  return (
    <div css={{ maxWidth: '100%', minWidth: 0 }}>
      {large && (
        <div
          css={{
            label: 'runway-popup-field__value--large',
            color: colours.white,
            fontSize: fontSize.large,
            letterSpacing: letterSpacing.small,
            lineHeight: '2.65rem'
          }}
        >
          {large}
        </div>
      )}
      {small && (
        <div
          css={{
            label: 'runway-popup-field__value--small',
            color: colours.white,
            fontSize: fontSize.body,
            whiteSpace: 'nowrap',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            maxWidth: '100%'
          }}
        >
          {small}
        </div>
      )}
    </div>
  );
}

ButtonValue.propTypes = {
  large: PropTypes.string,
  small: PropTypes.string
};

ButtonValue.defaultProps = {
  large: '',
  small: ''
};

function ButtonValues({ firstValue, secondValue }) {
  return (
    <div
      css={{
        display: 'flex',
        overflow: 'hidden',
        maxWidth: '100%'
      }}
    >
      <ButtonValue {...firstValue} />
      {secondValue && (
        <div
          css={{
            color: 'white',
            margin: '0 15px',
            fontSize: '36px',
            marginBottom: '20px'
          }}
        >
          -
        </div>
      )}
      {secondValue && <ButtonValue {...secondValue} />}
    </div>
  );
}

ButtonValues.propTypes = {
  firstValue: PropTypes.shape({
    large: PropTypes.string,
    small: PropTypes.string
  }).isRequired,
  secondValue: PropTypes.shape({
    large: PropTypes.string,
    small: PropTypes.string
  })
};

ButtonValues.defaultProps = {
  secondValue: null
};

export default ButtonValues;
