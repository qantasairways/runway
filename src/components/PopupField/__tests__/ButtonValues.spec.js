import React from 'react';
import { mount } from 'enzyme';
import ButtonValues from '../components/ButtonValues';

describe('ButtonValues', () => {
  let component;

  it('renders correctly with one large value', () => {
    const value = { large: 'MEL' };

    component = mount(<ButtonValues firstValue={value} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with one small value', () => {
    const value = { small: 'Melbourne, Australia' };

    component = mount(<ButtonValues firstValue={value} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with one large and small value', () => {
    const value = { large: 'MEL', small: 'Melbourne, Australia' };

    component = mount(<ButtonValues firstValue={value} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with two values', () => {
    const firstValue = { large: 'MEL', small: 'Melbourne, Australia' };
    const secondValue = { large: 'SYD', small: 'Sydney, Australia' };

    component = mount(
      <ButtonValues firstValue={firstValue} secondValue={secondValue} />
    );

    expect(component).toMatchSnapshot();
  });
});
