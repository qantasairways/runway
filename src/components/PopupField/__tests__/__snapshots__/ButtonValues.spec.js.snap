// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonValues renders correctly with one large and small value 1`] = `
.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

.emotion-2 {
  max-width: 100%;
  min-width: 0;
}

.emotion-0 {
  color: #fff;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.0625rem;
  -moz-letter-spacing: -0.0625rem;
  -ms-letter-spacing: -0.0625rem;
  letter-spacing: -0.0625rem;
  line-height: 2.65rem;
}

.emotion-1 {
  color: #fff;
  font-size: 1.125rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
}

<ButtonValues
  firstValue={
    Object {
      "large": "MEL",
      "small": "Melbourne, Australia",
    }
  }
  secondValue={null}
>
  <div
    className="emotion-3"
  >
    <ButtonValue
      large="MEL"
      small="Melbourne, Australia"
    >
      <div
        className="emotion-2"
      >
        <div
          className="emotion-0"
        >
          MEL
        </div>
        <div
          className="emotion-1"
        >
          Melbourne, Australia
        </div>
      </div>
    </ButtonValue>
  </div>
</ButtonValues>
`;

exports[`ButtonValues renders correctly with one large value 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

.emotion-1 {
  max-width: 100%;
  min-width: 0;
}

.emotion-0 {
  color: #fff;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.0625rem;
  -moz-letter-spacing: -0.0625rem;
  -ms-letter-spacing: -0.0625rem;
  letter-spacing: -0.0625rem;
  line-height: 2.65rem;
}

<ButtonValues
  firstValue={
    Object {
      "large": "MEL",
    }
  }
  secondValue={null}
>
  <div
    className="emotion-2"
  >
    <ButtonValue
      large="MEL"
      small=""
    >
      <div
        className="emotion-1"
      >
        <div
          className="emotion-0"
        >
          MEL
        </div>
      </div>
    </ButtonValue>
  </div>
</ButtonValues>
`;

exports[`ButtonValues renders correctly with one small value 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

.emotion-1 {
  max-width: 100%;
  min-width: 0;
}

.emotion-0 {
  color: #fff;
  font-size: 1.125rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
}

<ButtonValues
  firstValue={
    Object {
      "small": "Melbourne, Australia",
    }
  }
  secondValue={null}
>
  <div
    className="emotion-2"
  >
    <ButtonValue
      large=""
      small="Melbourne, Australia"
    >
      <div
        className="emotion-1"
      >
        <div
          className="emotion-0"
        >
          Melbourne, Australia
        </div>
      </div>
    </ButtonValue>
  </div>
</ButtonValues>
`;

exports[`ButtonValues renders correctly with two values 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

.emotion-2 {
  max-width: 100%;
  min-width: 0;
}

.emotion-0 {
  color: #fff;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.0625rem;
  -moz-letter-spacing: -0.0625rem;
  -ms-letter-spacing: -0.0625rem;
  letter-spacing: -0.0625rem;
  line-height: 2.65rem;
}

.emotion-1 {
  color: #fff;
  font-size: 1.125rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
}

.emotion-3 {
  color: white;
  margin: 0 15px;
  font-size: 36px;
  margin-bottom: 20px;
}

<ButtonValues
  firstValue={
    Object {
      "large": "MEL",
      "small": "Melbourne, Australia",
    }
  }
  secondValue={
    Object {
      "large": "SYD",
      "small": "Sydney, Australia",
    }
  }
>
  <div
    className="emotion-7"
  >
    <ButtonValue
      large="MEL"
      small="Melbourne, Australia"
    >
      <div
        className="emotion-2"
      >
        <div
          className="emotion-0"
        >
          MEL
        </div>
        <div
          className="emotion-1"
        >
          Melbourne, Australia
        </div>
      </div>
    </ButtonValue>
    <div
      className="emotion-3"
    >
      -
    </div>
    <ButtonValue
      large="SYD"
      small="Sydney, Australia"
    >
      <div
        className="emotion-2"
      >
        <div
          className="emotion-0"
        >
          SYD
        </div>
        <div
          className="emotion-1"
        >
          Sydney, Australia
        </div>
      </div>
    </ButtonValue>
  </div>
</ButtonValues>
`;
